linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_faillock_deny/oval/shared.xml

```diff
@@ -1,222 +1,290 @@
-    <criteria operator="AND" comment="Checks common to both scenarios">
-      <criterion test_ref="test_accounts_passwords_pam_faillock_preauth_silent_system-auth"
-      comment="pam_faillock.so preauth silent set in system-auth" />
-      <criterion test_ref="test_accounts_passwords_pam_faillock_account_phase_system-auth"
-      comment="pam_faillock.so set in account phase of system-auth" />
-      <criterion test_ref="test_accounts_passwords_pam_faillock_preauth_silent_password-auth"
-      comment="pam_faillock.so preauth silent set in password-auth" />
-      <criterion test_ref="test_accounts_passwords_pam_faillock_account_phase_password-auth"
-      comment="pam_faillock.so set in account phase of password-auth" />
-      <criteria operator="AND">
-        <criteria operator="OR" comment="system-auth">
-          <criterion test_ref="test_accounts_passwords_pam_faillock_numeric_default_check_system-auth"
-            comment="Perform check if pam_faillock authfail follows pam_unix even with lines skipped" />
-          <criterion test_ref="test_accounts_passwords_pam_faillock_authfail_deny_system-auth"
-            comment="Perform check if pam_faillock authfail follows pam_unix with either sufficient or default=ignore" />
+  <definition class="compliance" id="{{{ rule_id }}}" version="5">
+    {{{ oval_metadata("Lockout account after failed login attempts") }}}
+    <criteria operator="AND" comment="Check the proper configuration of pam_faillock.so">
+      <criteria operator="AND" comment="Check if pam_faillock.so is properly enabled">
+        <!-- pam_unix.so is a control module present in all realistic scenarios and also used
+             as reference for the correct position of pam_faillock.so in auth section. If the
+             system is properly configured, it must appear only once in auth section. -->
+        <criteria operator="AND" comment="Count occurrencies of pam_unix.so in system-auth">
+          <criterion test_ref="test_accounts_passwords_pam_faillock_deny_system_pam_unix_auth"
+                     comment="pam_unix.so appears only once in auth section of system-auth"/>
+          <criterion test_ref="test_accounts_passwords_pam_faillock_deny_password_pam_unix_auth"
+                     comment="pam_unix.so appears only once in auth section of password-auth"/>
         </criteria>
-        <criteria operator="OR" comment="password-auth">
-          <criterion test_ref="test_accounts_passwords_pam_faillock_numeric_default_check_password-auth"
-            comment="Perform check if pam_faillock authfail follows pam_unix even with lines skipped" />
-          <criterion test_ref="test_accounts_passwords_pam_faillock_authfail_deny_password-auth"
-          comment="pam_faillock.so authfail deny value set in password-auth" />
+
+        <!-- pam_faillock.so parameters can be defined directly in pam files or, in earlier
+             versions, in /etc/security/faillock.conf. The last is the recommended option when
```

Comment:
It's vice-versa, isn't it? Ie. They can be defined in /etc/security/faillock.conf or, in earlier versions, directly in pam files.

Reply:
Actually, newer versions. : ) Thanks @jan-cerny I am fixing.